Whats new 2.5 (#1481)
databricks/databricks-vscode: packages/databricks-vscode/resources/whats-new/2.5.md

<img src="./1.2/databricks-logo.png" alt="databricks-logo" width="100"/>

# Databricks Extension v2.5.0

## What's new?

> -   [Validation and partial update actions for pipelines](#validation)

Reviewer: ```suggestion
-   [Pipeline validation and partial update actions](#validation)
```

-   [Pipeline events and diagnostics](#events)

> -   [Pipeline datasets and their schema](#datasets)

Reviewer: ```suggestion
-   [Pipeline datasets and their schemas](#datasets)
```
Do we need "and their schemas" here? Isn't supporting datasets assume schemas are included?

Author: Will remove it

-   [Quick run action](#quick-run)

> ### <a id="validation"></a> Validation and partial update actions for pipelines

Reviewer: ```suggestion
### <a id="validation"></a> Pipeline validation and partial update actions
```

> In previous versions, you could only trigger a “refresh all” pipeline update. Now we offer validation and partial update actions:

Reviewer: ```suggestion
In previous versions, you could only trigger a [Refresh all](https://docs.databricks.com/en/delta-live-tables/updates.html#how-dlt-updates) pipeline update. Now you can do validation and partial updates:
```
How about this way to add text treatment (instead of quotes)?

Reviewer: (Also fine to just leave it with the quotes but the second sentence should be second person.

> In previous versions, you could only trigger a “refresh all” pipeline update. Now we offer validation and partial update actions:

Author: full update has another meaning - https://docs.databricks.com/en/delta-live-tables/updates.html#how-delta-live-tables-updates-tables-and-views

There's "refresh all" and "full refresh all"

> In previous versions, you could only trigger a “refresh all” pipeline update. Now we offer validation and partial update actions:

Reviewer: ah ok

<img src="./2.5/validation.png" alt="validation action" width="600"/>

> In addition to that, the partial update action preloads datasets from past runs, lets you type in table names manually, or select the full-refresh option:

Reviewer: ```suggestion
The partial update action preloads datasets from past runs, and lets you enter table names or select a full refresh:
```

<img src="./2.5/partial-update.png" alt="partial update action" width="600"/>

### <a id="events"></a> Pipeline events and diagnostics

> During and after a pipeline update, the extension now shows events in the resource explorer UI:

Reviewer: ```suggestion
During and after a pipeline update, the extension now shows events in the resource explorer UI:
```

<img src="./2.5/events.png" alt="pipeline events" width="600"/>

> Any warnings or errors that have source locations are also displayed in VSCode’s Problems panel and highlighted in the source files themselves:

Reviewer: ```suggestion
Any warnings or errors that have source locations are also displayed in the Visual Studio Code **PROBLEMS** panel and highlighted in the source files:
```
Weird, but that's how VSCode docs write about that problems panel.

<img src="./2.5/diagnostics.gif" alt="pipeline diagnostics" width="600"/>

> ### <a id="datasets"></a> Pipeline datasets and their schema

Reviewer: ```suggestion
### <a id="datasets"></a> Pipeline datasets and their schemas
```
(Again, maybe we could drop "and their schemas"?)

You can now see pipeline datasets and their schema:

<img src="./2.5/datasets.png" alt="pipeline datasets" width="600"/>

Datasets are gathered from the events of past runs, so make sure to update or validate your pipeline at least once.

### <a id="quick-run"></a> Quick run action

> All run and update actions used to be tied to the resource explorer UI, but now you can also use a "deploy the bundle and run a resource" command from the command palette or with a keyboard shortcut:

Reviewer: ```suggestion
Previously, run and update actions were only available from the extension resource explorer UI. Now you can also use a new `Deploy the bundle and run a resource` command from the Command Palette or with a keyboard shortcut:
```

<img src="./2.5/run-command-search.png" alt="pipeline datasets" width="600"/>

> By default, we will ask you what resource you want to execute:

Reviewer: ```suggestion
By default, you select the resource to execute:
```

<img src="./2.5/run-command-ui.png" alt="pipeline datasets" width="600"/>

You can also specify a shortcut to run the command with specific arguments.

> Open keyboard shortcuts JSON (execute this command `Preferences: Open Keyboard Shortcuts (JSON)`). Add a new entry to the `keybindings` array:

Reviewer: ```suggestion
Open the keyboard shortcuts JSON using the command `Preferences: Open Keyboard Shortcuts (JSON)`. Add a new entry to the `keybindings` array:
```
